### PR TITLE
Broadcom AFBR-S50 Distance Sensor Driver: Test API

### PR DESCRIPTION
**Pull request by samchamberlin**
**Describe problem solved by this pull request**
This PR for the Broadcom AFBR-S50 Distance Sensor driver enables execution of a series of tests in order to verify the HAL implementation.

**Describe your solution**
The "test" arguement can be used within nutt shell for the afbrs50 driver in order to run the series of tests. Updates are printed out in the shell and delays were included to prevent breakpoint traps & faults.

**Test data / coverage**
The additions to the driver were tested successfully within the nutt shell on macOS and Linux.

**Additional context**
Screen Shot 2021-06-28 at 2 33 20 PM

### CODE REVIEW

File: boards/ark/can-flow/init/rc.board_sensors [hunk 1]
```diff
@@ -8,4 +8,4 @@ paw3902 -s start -Y 180
 bmi088 -A -s -R 4 start
 bmi088 -G -s -R 4 start
 
-afbrs50 start
+# afbrs50 start
```
Comment (AlexKlimaj) [inline boards/ark/can-flow/init/rc.board_sensors:11]: Revert this file. We want the driver to start normally.

File: src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp [hunk 2]
```diff
@@ -137,36 +167,69 @@ int AFBRS50::init()
...
+>>>>>>> 1b4edf7757 (update)
+			_px4_rangefinder.set_min_distance(_min_distance);
+			_px4_rangefinder.set_max_distance(_max_distance);
+>>>>>>> cd4b5e51b4 (parent bda36a021903ff66cd9ce04b7cd66202993ebba6)
+=======
+			min_distance = 0.08f;
+			max_distance = 30.f;
+			_px4_rangefinder.set_min_distance(min_distance);
+			_px4_rangefinder.set_max_distance(max_distance);
+>>>>>>> 3864c90f95 (added gate for processing distance measurements)
```
Comment (AlexKlimaj) [inline src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp]: I think you have merge conflicts that need to be resolved.

File: src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp [hunk 3]
```diff
@@ -137,36 +167,69 @@ int AFBRS50::init()
...
 			_px4_rangefinder.set_fov(math::radians(6.f));
 			PX4_INFO_RAW("AFBR-S50LV85D (v1)\n");
 			break;
 
 		case AFBR_S50MV68B_V1:
-			_px4_rangefinder.set_min_distance(0.08f);
-			_px4_rangefinder.set_max_distance(10.f);
+			_min_distance = 0.08f;
+			_max_distance = 10.f;
+			_px4_rangefinder.set_min_distance(_min_distance);
```
Comment (AlexKlimaj) [inline src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp:168]: Is this from your other PR?

Comment (AlexKlimaj) [reply src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp:168]: I think you need to rebase this PR from master.

**Review by AlexKlimaj — CHANGES_REQUESTED** [review]

**Comment by dagar** [comment]
Awesome, I wasn't sure if we were going to pass the timing tests.

**Comment by AlexKlimaj** [comment]
@samchamberlin run a "make format" and repush.

File: src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp [hunk 4]
```diff
@@ -294,6 +311,25 @@ static int stop()
 	return PX4_OK;
 }
 
+static int test(const uint8_t rotation)
```
Comment (mcsauder) [inline src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp:314]: These changes all look good to me, I just might be missing why this method needs to be static.

**Pull request closed without merging**